src/git/http.ts

```diff
@@ -398,12 +399,18 @@ async function* generatePackData({
   objectIds: Array<ObjectId>;
   chunkSize?: number;
 }): AsyncGenerator<Buffer, void, void> {
-  const packFile = await git.packObjects({
-    fs: efs,
-    dir,
-    gitdir: gitDir,
-    oids: objectIds,
-  });
+  let packFile: PackObjectsResult;
+  try {
+    packFile = await git.packObjects({
+      fs: efs,
+      dir,
+      gitdir: gitDir,
+      oids: objectIds,
+    });
+  } catch (e) {
+    // Return without sending any data
+    return;
+  }
   if (packFile.packfile == null) utils.never('failed to create packFile data');
   let packFileBuffer = Buffer.from(packFile.packfile.buffer);
```

Comment:
The `never` should be reserved for programming errors. If it can happen, it should be an exception.

Reply:
I mis-remembered, the `never` here is just to enforce the fact that `packfile` is defined. By our usage it should always be defined. So the `never` is proper here, just the message is off point.

```ts
    /**
     * - The packfile contents. Not present if `write` parameter was true, in which case the packfile was written straight to disk.
     */
    packfile?: Uint8Array;
};
```

Reply:
I made some comments in the code to clarify things, Also removed the never for a `return` to align with the above catch block.